docs/javascript-unit-tests.md

```diff
@@ -0,0 +1,712 @@
+[read about atomic components in cfgov-refresh](../atomic-structure/).
+
+!!! note "Child apps"
+    If you’re working on something in a child app,
+    put it in `test/unit_test/appname/js/`.
+    Otherwise, if you’re working on something that
+    belongs to cfgov-refresh generally,
+    it should go in the corresponding folder under `test/unit_test/js/`.
+
+Now that you have your sample JS and test files in the right places,
+let’s try running them and see what happens!
+I’ll refer to `sample-spec.js` and `sample.js` in the instructions below,
+but you should work in your own new test file and JavaScript file
+to save and commit your changes.
+
+1. Edit line 6 of your spec file and remove the `.skip` method.
+   The line should now read:
+   ```js
+   it( 'should return a string with expected value', () => {
+     …
+   } );
+   ```
+1. Run your sample test using `gulp test:unit --specs=js/modules/sample-spec.js`
+   (substituting your own filename).
+   The test should fail – this is expected.
+   Remember, when doing TDD, we want to write our test to fail first,
+   then write the corresponding JavaScript that will make the test pass.
+1. Make the test pass by changing your script’s line 7
+   ([see `sample.js`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/js/modules/sample.js))
+   to the following:
+   ```js
+   return 'Shredder';
+   ```
+1. Run the test again to confirm the test now passes.
+   Doesn’t it feel good?
+
+[See the “Running tests” section of this page](#running-tests)
+for additional commands to run tests.
+
+
+### File structure (basic layout of a test file)
+
+In order to make the `sample-spec.js` more meaningful to your own use case,
+you’ll need to know how to structure a unit test using Jest methods.
+Let’s take a look at the structure of our very basic sample test file.
+
+#### Loading dependencies
+
+Line 1 of any spec file will use an `import` statement to
+include the JavaScript file that you are testing.
+Additional dependencies should be added in the same manner.
+
+```js
+import * as sample from '../../../../cfgov/unprocessed/js/modules/sample.js';
+```
+
+Some test files use `const` declarations to
+`require` scripts instead of `import`,
+because those files were written before `import` was available.
+We prefer to use `import` because it allows for
+[tree shaking in Webpack](https://webpack.js.org/guides/tree-shaking/),
+meaning if two modules are importing the same module
+it should only be included in the bundle once,
+whereas with `require` it would be included twice.
+
+A consequence is that variables can't be used in the import path,
+as they prevent Webpack from figuring out which modules are duplicates.
+For example, this snippet shows how a `require` statement
+should be converted to an `import` statement,
+but without including the `BASE_JS_PATH` variable in the file path:
+
+```js
+// works but could duplicate footer-button.js if other files also require it
```

Comment:
TRIVIAL: Follow grammatical rules in comments, so capitalization in punctuation.